### DISCUSSION TITLE
Size of DLT log file and filter option

### DISCUSSION THREAD

**Discussion by egurbas** [COVESA/dlt-viewer#604] [Q&A]
Dear,
even if i apply the filtering settings, on the background DLT Viewer keeps loading all the logs into dlt log file. 
Yes, you can see only the filtered ones in the screen / window, but when you terminate DLT Viewer, you can see them all in the file.

**Note:** I am using CLI option to run DLT Viewer for automation purposes.
e.g. `dlt-viewer.exe -s -p <project_file_dlp> -f <filter_file.dlf> -l <output_file.dlt>`

**Comment by alexmucde** [comment]
You need to use the export feature per commandline to generate a new DLT file including only the filtered messages.
You can use the option "-c new.dlt -dd". You can also export to txt files "-c new.txt" or csv files "-c new.csv --csv".
Please check all the command line options with "dlt-viewer --help".

**Reply by egurbas** [reply]
actually it is what I did. After starting DLT Viewer with the options that I listed above (in my question), I terminate it and send the below command:
`dlt-viewer.exe -s -c <dlt_log_file.dlt> <dlt_log_file.txt>`
and as I described in above problem statement, filter option applies only to GUI of Dlt Viewer, not the *.dlt file itself.

**Reply by egurbas** [reply]
do you mean to try
```
dlt-viewer.exe -s -p <project_file_dlp> -f <filter_file.dlf> -l <output_file.dlt>

testWaitForTimeout (<some time>);
sysExec("taskkill", "/F /IM dlt-viewer.exe");  // log file will be saved, only after termination of DLR Viewer application

dlt-viewer.exe -s -c <output_file.dlt> -dd <output_file.txt>  // option -dd is added
```

**Reply by alexmucde** [reply]
This should work, -c should follow the output filename:
dlt-viewer.exe -s -c <output_file.dlt> -dd <input_file.dlt>  // option -dd is added

**Reply by egurbas** [reply]
Thanks for your support :) 
I will try and share my result in here for the community.

**Reply by egurbas** [reply]
hi Alex
I tried here is my observations:
```
dlt-viewer.exe -s -p <project_file_dlp> -f <filter_file.dlf> -l <output_file.dlt>

testWaitForTimeout (<some time>);
sysExec("taskkill", "/F /IM dlt-viewer.exe");  // log file will be saved, only after termination of DLR Viewer application
```
after the code lines above, again huge amount of data in output_file_file.dlt is being generated. 
`dlt-viewer.exe -s -c <output_file.dlt> -dd <output_file.txt>  // option -dd is added`
and after above code lines, no file conversion took place. Actually option -dd is meant for it.

so, your recommendation did not solve the issue. Anyway, thanks for your time.

**Reply by egurbas** [reply]
and by the way, I also used -dd option every corner of the CLI commands, e.g.
`dlt-viewer.exe -s -dd -p <project_file_dlp> -f <filter_file.dlf> -l <output_file.dlt>`
actually -dd stands for "**_conversion will NOT be done, as ve as decoded messages in dlt format_**"

**Reply by alexmucde** [reply]
You are always using the wrong order of the parameters.  Use them exactly as i have told you first:
dlt-viewer.exe -s -dd -c <output_file.dlt> <input_file.dlt>
After -c must follow the output file name, like in this use case a DLT file. And the you can add the input DLT file.
PLease use the parameters exactly as described in the help (--help).

**Reply by egurbas** [reply]
> This should work, -c should follow the output filename: dlt-viewer.exe -s -c <output_file.dlt> -dd <input_file.dlt> // option -dd is added

This above is what you proposed. And, out of courtesy, this is exactly what i tried and also I tried different variations as well, and I must say, it is time consuming. Because, if you follow my original problem statement, I NEED TO launch DTL VIEWER in such a way that the filter file that I provided in the CLI command should generate *.dlt log file containing only the ones passing the filter.

And again as for your proposal, I need to update my second CLI command which is about conversion. Moreover, if you look at --help, you will see that -dd stands for "conversion will NOT be done, as decoded messages in dlt format", which is nonsense and not related my problem statement.

So, if you want to contribute, please FIRST try what you propose. Otherwise it is just time consuming.

**Comment by alexmucde** [comment]
I think we are still not talking about the same thing. The use case you are describing is not working at once. You always need two steps.

First one filter the DLT file <input_file.dlt> using a filter <filter_file.dlf> and create a new one <output_file.dlt>, which contains only the filtered messages everything by command line:
dlt-viewer.exe -s -t -dd -c output_file.dlt input_file.dlt filter_file.dlf

You can also use the new dlt-commander:
dlt-commander.exe -dd -c output_file.dlt input_file.dlt filter_file.dlf

Second step then open the new output_file.dlt in the DLT Viewer.

There is a limitation with -dd, it works only for verbose messages.
Important is which version you are using, the command lin options have changed a little bit. I am talking always about latest 2.27.0 version.

**Comment by EthemGurbas** [comment]
> Dear, even if i apply the filtering settings, on the background DLT Viewer keeps loading all the logs into dlt log file. Yes, you can see only the filtered ones in the screen / window, but when you terminate DLT Viewer, you can see them all in the file.
> 
> **Note:** I am using CLI option to run DLT Viewer for automation purposes. e.g. `dlt-viewer.exe -s -p <project_file_dlp> -f <filter_file.dlf> -l <output_file.dlt>`

ok, once again, I try to simply put the steps as below: 

**Step 1 :** open DLT Viewer, make it listen incoming DLT messages from ECU but only for filtered ones
**Step 2 :** generate an event and correspoding DLT messages on the client side (DLT Viewer) _--> out of scope for our discussion_
**Step 3 :** stop DLT viewer, to generate the log file (*.dlt) on the file system of host (e.g. TestBench PC)
**Step 4 :** convert dlt type log file into human readable one (e.g. *.txt)
**Step 5 :** by using file commands of CAPL, read human readable DLT log file and search for the expected log message to provide your verdict. --> out of scope for our discussion

It means that I am **_not opening DLT Viewer again_** to view the filtered messages, as you suggested above.

**Reply by vifactor** [reply]
@EthemGurbas do I understand correctly that your expectation is that filtering should happen on the **Step 1** and the issue is that it indeed happens in the UI, but not for the stored dlt file?

**Reply by EthemGurbas** [reply]
@vifactor yes

**Reply by vifactor** [reply]
Please check if you have this checkbox enabled:
![image](https://github.com/user-attachments/assets/453bd0f0-21ba-46d7-b1ab-683bd00d486f)

**Reply by EthemGurbas** [reply]
checked all settings already, including this one. It is already enabled.

**Reply by EthemGurbas** [reply]
and for automation purposes, the first one ("Auto connect to ECUs at start") is also enabled. So, whenever DLT Viewer is launced, it connect and begin to listed the port for incoming log messages.

**Reply by vifactor** [reply]
I tried to do reproduce your observations. I use v2.26.2 (I guess you have that?), but on Linux. And I must say that I have issues with persistence of the settings above between dlt-viewer invokations, in the mean time the filtering actually happens for me (both in UI and when stored to file) as soon as I have that "Logging only filtered DLT Messages" enabled. Maybe it is a Windows specific thing?.. For me unfortunately, it would be not easy to try on Windows.

**Reply by EthemGurbas** [reply]
@vifactor my verison is 2.23.0 and it is running on Windows
I will update it with the lates one (2.27) and will share my observations again in here. Thanks

**Reply by vifactor** [reply]
I am puzzled then... the setting "Logging only filtered DLT Messages" appeared between 2.25 and 2.26 as far as I can see in the code, unless it was in a different place of the settings dialog in 2.23:
![image](https://github.com/user-attachments/assets/d8fa0fdb-bcc7-4b10-bb9e-5ebab65cbd62)

**Reply by EthemGurbas** [reply]
Sorry, you are right. I am confused with the settings of "filter" itself. 
I will try the latest one.

**Reply by EthemGurbas** [reply]
Under "Filter", there is an options called "Filters Enabled". This is the one that I am confused with. Under project settings, there is no such check box at my current version.
